src/pymor/algorithms/basic.py

```diff
@@ -67,3 +68,37 @@ def almost_equal(U, V, product=None, norm=None, rtol=1e-14, atol=1e-14):
     ERR_norm = norm(X)
 
     return ERR_norm <= atol + V_norm * rtol
+
+
+def relative_error(U, V, product=None):
+    """Compute error between U and V relative to norm of U."""
+    return (U - V).norm(product) / U.norm(product)
+
+
+def project_array(U, basis, product=None, orthonormal=True):
+    """Orthogonal projection of |VectorArray| onto subspace.
+
+    Parameters
+    ----------
+    U
+        The |VectorArray| to project.
+    basis
+        |VectorArray| of basis vectors for the subspace onto which
+        to project.
+    product
+        Inner product |Operator| w.r.t. which to project.
+    orthonormal
+        If `True`, the vectors in `basis` are assumed to be orthonormal
+        w.r.t. `product`.
+
+    Returns
+    -------
+    The projected |VectorArray|.
+    """
+    if orthonormal:
+        return basis.lincomb(U.inner(basis, product))
+    else:
+        gramian = basis.gramian(product)
+        rhs = basis.inner(U, product)
+        coeffs = scipy.linalg.solve(gramian, rhs, sym_pos=True, overwrite_a=True, overwrite_b=True).T
```

Comment:
`sym_pos=True` should probably be replaced by `assume_a='pos'` (see [here](https://github.com/scipy/scipy/blob/a8229ce213ea7731afb4a4a91e4681a0821ba40b/scipy/linalg/basic.py#L73-L76)). But actually, is `gramian` guaranteed to be numerically symmetric and positive definite? Adding `gramian = (gramian + gramian.T) / 2` might be a good idea, and `solve` will probably detect if `gramian` is close to being singular.

Reply:
I used `sym_pos` instead of `assume_a` since the latter is only present in the latest SciPy release and I didn't want to make the code more complicated. - I guess in presence of a `product`, there is no guarantee that `gramian` is exactly spd. However, why do you think, that `gramian = (gramian + gramian.T) / 2` would improve the result of the `solve` call in that case?

Reply:
I guess it's mostly for my own sanity. But, it's nice that (A+A^T)/2 is the closest symmetric matrix to A. Anyway, Cholesky factorization can be computed in a stable way, so there shouldn't be a huge difference. Probably the bigger issue is that the condition number of `gramian` is the square of the condition number of `basis` (at least in the standard inner product).
Anyhow, calling `gram_schmidt` before avoids all of this, so feel free to ignore my previous comments.

Reply:
Ok, I would also doubt that A+A^T/2 has an important effect. Merging as is ..